Automattic/wp-calypso
# Importers Flux store: remove api actions

We don't need no `apiStart`, `apiSuccess` and `apiFailure` actions. Receiving the importer state update and marking the state as hydrated is sufficiently covered by the `IMPORTS_IMPORT_RECEIVE` action. This PR removes them.

I had to be careful about implementing the `IMPORTS_IMPORT_RECEIVE` action reducer correctly. It needs to:
- always mark the state as hydrated no matter what the successful response contains, like empty state or a defunct state that should be ignored.
- not update importer state if it's locked, but still mark as hydrated in that case
- not try to convert empty state (`[]`) with `fromApi`. That function assumes some properties always exist and that assumption is right for non-empty states

In the `fetchImporterState` handler, I removed the
```js
castArray( data ).forEach( ( importerData ) => {
  receiveImporterStatus( fromApi( importerData ) );
} );
```
code and changed it to simply:
```js
receiveImporterStatus( data );
```
There is always only one importer status returned, it's never an array of multiple ones. The empty array response (`[]`) is a quirk of the PHP REST implementation, which returns an empty PHP array (`return array()`) which is then serialized into empty array rather than empty object.

The `castArray` call just causes that `receiveImporterStatus` is not called for the response 🙂 And it's also the reason why the `isHydrated` reducer had to react to `IMPORTS_FETCH_COMPLETED` action, too. Now it has to react only to `IMPORTS_IMPORT_RECEIVE`.

**How to test:**
I added a new unit test for the locked case, so running the unit tests is one way to test this.

Other than that, verify that the list of importers is correctly displayed at `/import/:site`, first as inactive, then as active after REST response confirms there is no active running import.

Also verify that if an import is running in the background, it's status is correctly loaded and displayed on the `/import/:site` page reload.
